tensorflow-examples/src/main/java/org/tensorflow/model/examples/mnist/training/LinearRegressionExample.java

```diff
@@ -0,0 +1,154 @@
+ */
+public class LinearRegressionExample {
+    /**
+     * Amount of data points.
+     */
+    private static int N = 10;
+
+    /**
+     * This value is used to fill the Y placeholder in prediction.
+     */
+    private static final float NO_MEANING_VALUE_TO_PUT_IN_PLACEHOLDER = 2000f;
+
+    public static void main(String[] args) {
+        // Prepare the data
+        float[] xValues = {1f, 2f, 3f, 4f, 5f, 6f, 7f, 8f, 9f, 10f};
+        float[] yValues = new float[10];
+
+        Random rnd = new Random(42);
+
+        for (int i = 0; i < yValues.length; i++) {
+            yValues[i] = (float) (10 * xValues[i] + 2 + 0.1 * (rnd.nextDouble() - 0.5));
+        }
+
+        try (Graph graph = new Graph()) {
+            Ops tf = Ops.create(graph);
+
+            // Define placeholders
+            Placeholder<TFloat32> X = tf.placeholder(TFloat32.DTYPE, Placeholder.shape(Shape.scalar()));
+            Placeholder<TFloat32> Y = tf.placeholder(TFloat32.DTYPE, Placeholder.shape(Shape.scalar()));
+
+            // Define variables
+            Variable<TFloat32> weight = tf.variable(Shape.scalar(), TFloat32.DTYPE);
+            Variable<TFloat32> bias = tf.variable(Shape.scalar(), TFloat32.DTYPE);
+
+            // Init variables
+            Assign<TFloat32> weightInit = tf.assign(weight, tf.constant(1f));
+            Assign<TFloat32> biasInit = tf.assign(bias, tf.constant(1f));
+
+            // Define the model function weight*x + bias
+            Mul<TFloat32> mul = tf.math.mul(X, weight);
+            Add<TFloat32> yPredicted = tf.math.add(mul, bias);
+
+            // Define loss function MSE
+            Pow<TFloat32> sum = tf.math.pow(tf.math.sub(yPredicted, Y), tf.constant(2f));
+            Div<TFloat32> mse = tf.math.div(sum, tf.constant(2f * N));
+
+            // Define gradient operators
+            List<Variable<TFloat32>> variables = new ArrayList<>();
+            variables.add(weight);
+            variables.add(bias);
+
+            Gradients gradients = tf.gradients(mse, variables);
+
+            Constant<TFloat32> alpha = tf.constant(0.2f);
+
+            ApplyGradientDescent<TFloat32> weightGradientDescent = tf.train.applyGradientDescent(weight, alpha, gradients.dy(0));
+            ApplyGradientDescent<TFloat32> biasGradientDescent = tf.train.applyGradientDescent(bias, alpha, gradients.dy(1));
+
+            try (Session session = new Session(graph)) {
+                // Initialize graph variables
+                session.runner()
+                        .addTarget(weightInit)
+                        .addTarget(biasInit)
+                        .run();
+
+                // Train the model on data
+                for (int i = 0; i < xValues.length; i++) {
+                    float y = yValues[i];
+                    float x = xValues[i];
+
+                    Tensor<TFloat32> xTensor = TFloat32.scalarOf(x);
+                    Tensor<TFloat32> yTensor = TFloat32.scalarOf(y);
```

Comment:
When allocating tensors, you need to protect their resources in a try block, like this:
```java
try (Tensor<TFloat32> xTensor = TFloat32.scalarOf(x);
    Tensor<TFloat32> yTensor = TFloat32.scalarOf(y)) {
 ...
}
```
I know it is not pretty but that is required to avoid any potential memory leaks and our examples should show this